src/ke.cpp

```diff
@@ -620,6 +620,25 @@ class _usersim_emulated_dpc
     bool terminate;
 };
 
+NTSTATUS
+KeGetProcessorNumberFromIndex(ULONG ProcessorIndex, _Out_ PPROCESSOR_NUMBER ProcNumber)
+{
+    if (ProcessorIndex >= GetMaximumProcessorCount(ALL_PROCESSOR_GROUPS)) {
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    ProcNumber->Group = (WORD)(ProcessorIndex / 64);
+    ProcNumber->Number = (UCHAR)(ProcessorIndex % 64);
+    ProcNumber->Reserved = 0;
+    return STATUS_SUCCESS;
+}
+
+ULONG
+KeGetProcessorIndexFromNumber(_In_ PPROCESSOR_NUMBER ProcNumber)
+{
+    return (ULONG)ProcNumber->Group * 64 + (ULONG)ProcNumber->Number;

```

Comment:
It's unfortunately much more complicated than this. Groups may have 40 processors for instance, in an 80 processor machine.

Reply:
See msquic code here: https://github.com/microsoft/msquic/blob/main/src/platform/platform_winuser.c#L81

Reply:
Should be fixed now.